src/unix/tty.c

```diff
@@ -231,6 +231,15 @@ uv_handle_type uv_guess_handle(uv_file file) {
       return UV_UDP;
 
   if (type == SOCK_STREAM) {
+#if defined (_AIX)
+  /* on AIX the gettsockname call returns an empty sa structure
+   * for sockets of type AF_UNIX.  For all other types it will
+   * return a properly filled in structure.
+   */
+  if (len == 0)
+    return UV_NAMED_PIPE;
```

Comment:
Indent if off here, there's a typo in the comment (s/gettsockname/getsockname/) and can you write the guard as `#if defined(_AIX)`?  LGTM apart from that.